pallets/frequency-tx-payment/src/lib.rs

```diff
@@ -423,12 +443,27 @@ where
 		info: &DispatchInfoOf<<T as frame_system::Config>::RuntimeCall>,
 		len: usize,
 	) -> Result<(BalanceOf<T>, InitialPayment<T>), TransactionValidityError> {
+		use crate::types::GetAddKeyData;
 		match call.is_sub_type() {
 			Some(Call::pay_with_capacity { call }) =>
 				self.withdraw_capacity_fee(who, &vec![*call.clone()], len),
 			Some(Call::pay_with_capacity_batch_all { calls }) =>
 				self.withdraw_capacity_fee(who, calls, len),
-			_ => self.withdraw_token_fee(who, call, info, len, self.tip(call)),
+			_ => {
+				if let Some((owner_account_id, msa_id)) = T::MsaCallFilter::get_add_key_data(call) {
+					if T::MsaKeyProvider::key_eligible_for_free_addition(
+						owner_account_id.clone(),
+						msa_id,
+					) {
+						let tip = self.tip(call);
+						let fee = pallet_transaction_payment::Pallet::<T>::compute_fee(
```

Comment:
Unsure if this should be forced to zero.

Reply:
I think we can test it maybe manually to see what happens with the fee or with just zero. The idea is that the caller should not have any token and it should work.
The other question that comes up is how about the capacity? Should it be free also if called with capacity? My impression was that since these are going to be called by custodial wallet under the hood we probably want to make it free only for capacity calls.

Reply:
Is the custodial wallet a Provider? I think maybe not...? They shouldn't have to be a Provider, they currently don't actually execute any calls on-chain directly, so for the cutover it might be better to have this be free for token-based calls, not Capacity.

Reply:
NB for future us:  they are a Provider, they don't normally make capacity calls, but we decided to make this a subsidized capacity call for better control over who's calling & spam prevention.